### PR TITLE
Adjust CUDA memory leak test

### PR DESCRIPTION
**Pull request by ilia-cher**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40504 Adjust CUDA memory leak test**

Summary:
Make CUDA mem liek test not flaky

Test Plan:
python test/test_profiler.py

Differential Revision: [D22215527](https://our.internmc.facebook.com/intern/diff/D22215527)

### CODE REVIEW

File: test/test_profiler.py [hunk 1]
```diff
@@ -33,13 +33,15 @@ def test_mem_leak(self):
...
-        # using much smaller threshold but not zero to reduce flakiness
-        self.assertTrue(max_diff < 100 * 1024)
+        # with CUDA events leaking the increase in memory was ~7 MB between
+        # profiler invocations above
+        is_increasing = True
+        for idx in range(len(last_rss)):
+            if idx == 0:
+                continue
+            is_increasing = is_increasing and (last_rss[idx] > last_rss[idx - 1])
+        self.assertTrue(not is_increasing,
```
Comment (malfet) [inline test/test_profiler.py:43]:
```suggestion
        self.assertTrue(is_increasing,
```

**Review by malfet — APPROVED** [review]

File: test/test_profiler.py [hunk 2]
```diff
@@ -33,13 +33,15 @@ def test_mem_leak(self):
...
-        # with CUDA events leaking the increase in memory was ~7 MB,
-        # using much smaller threshold but not zero to reduce flakiness
-        self.assertTrue(max_diff < 100 * 1024)
+        # with CUDA events leaking the increase in memory was ~7 MB between
+        # profiler invocations above
+        is_increasing = True
+        for idx in range(len(last_rss)):
+            if idx == 0:
+                continue
+            is_increasing = is_increasing and (last_rss[idx] > last_rss[idx - 1])
```
Comment (malfet) [inline test/test_profiler.py:42]:
```suggestion
        is_increasing = all([last_rss[idx] > last_rss[idx-1] for idx in range(1, len(last_rss))])
```

File: test/test_profiler.py (hunk 1 above)
Comment (ilia-cher) [reply test/test_profiler.py:43]: the condition should be that it is not increasing

Comment (malfet) [reply test/test_profiler.py:43]: Hmm, then there are definitely a leak:
```
 memory usage is increasing, deque([1951924224, 1951936512, 1951952896, 1951977472, 1951985664], maxlen=5)
```

**Pull request closed without merging**